src/openms/source/QC/Ms2IdentificationRate.cpp

```diff
@@ -0,0 +1,159 @@
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow$
+// $Authors: Patricia Scheil, Swenja Wagner$
+// --------------------------------------------------------------------------
+
+//#include <OpenMS/QC/Ms2IdentificationRate.h>
+#include <include/OpenMS/QC/Ms2IdentificationRate.h>
+#include <algorithm>
+#include <iostream>
+
+
+
+namespace OpenMS
+{
+
+  Ms2IdentificationRate::Ms2IdentificationRate() = default;
+  Ms2IdentificationRate::~Ms2IdentificationRate() = default;
+
+  Int64 Ms2IdentificationRate::countPeptideId_(std::vector<PeptideIdentification> peptide_id, bool force_fdr)
+  {
+    Int64 counter{};
+
+    counter += count_if(peptide_id.begin(), peptide_id.end(), [force_fdr](PeptideIdentification const & x)
+    {
+
+      if ( !x.getHits().empty() )
+      {
+        if (x.getHits()[0].metaValueExists("target_decoy") && !force_fdr)
+        {
+          if (x.getHits()[0].getMetaValue("target_decoy") == "target")
+          {
+            return true;
+          }
+          else
+          {
+            return false;
+          }
+        }
+        else if (force_fdr)
+        {
+          return true;
+        }
+        else
+        {
+          throw Exception::Precondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "FDR was not made. If you want to continue without FDR use -force_fdr");
+        }
+      }
+      else
+      {
+        LOG_WARN << "Ms2IdentificationRate: There is a Peptideidentification without PeptideHits." << "\n";
+        return false;
+      }
+    }
+    );
+    return counter;
+  }
+
+
+  //computes number of peptide identifications, number of ms2 spectra and ratio
+  //data is stored in vector of structs
+//  void Ms2IdentificationRate::compute(FeatureMap const & feature_map, MSExperiment const & exp, std::string file, bool force_fdr)
+  void Ms2IdentificationRate::compute(FeatureMap const & feature_map, MSExperiment const & exp, bool force_fdr)
+  {
+    //checks if data exists
+      if (feature_map.empty())
+      {
+        throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "FeatureXML is corrupted or empty");
```

Comment:
if the file were corrupted, that piece of code would never be reached. Dont spread rumors :)

if its empty, thats fine... idrate = 0